streaming/java/streaming-state/src/main/java/org/ray/streaming/state/backend/AbstractStateBackend.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ray.streaming.state.backend;
+
+import static org.ray.streaming.state.config.ConfigKey.DELIMITER;
+
+import java.io.Serializable;
+import java.util.Map;
+import org.ray.streaming.state.backend.impl.MemoryStateBackend;
+import org.ray.streaming.state.config.ConfigKey;
+import org.ray.streaming.state.keystate.KeyGroup;
+import org.ray.streaming.state.keystate.desc.AbstractStateDescriptor;
+import org.ray.streaming.state.serde.IKMapStoreSerDe;
+import org.ray.streaming.state.store.IKMapStore;
+import org.ray.streaming.state.store.IKVStore;
+
+/**
+ * AbstractStateBackend. This class is state backend factory.
```

Comment:
This class is definitely not state backend factory. It just contains several static factory methods.

For document, it might be helpful we document what is a state backend, or we just omit it for now.

Reply:
Thanks for the comments. I'll try to fix the naming and document.